exporters/pelorus/deserialization/__init__.py

```diff
@@ -79,54 +83,119 @@
     MissingFieldError,
     TypeCheckError,
 )
-from pelorus.utils import BadAttributePathError, get_nested
+from pelorus.utils import BadAttributePathError, format_path, get_nested, split_path
 from pelorus.utils._attrs_compat import NOTHING
 
 # metadata
 _NESTED_PATH_KEY = "__pelorus_structure_nested_path"
+_RETAIN_SOURCE_KEY = "__pelorus_structure_retain_source"
 
 T = TypeVar("T")
 
 PRIMITIVE_TYPES: tuple[type, ...] = (int, float, str, bool)
 
 
-def nested(path: str) -> dict[str, Any]:
-    "Put the nested data path in a metadata dict."
-    return {_NESTED_PATH_KEY: path}
+def nested(path: Union[str, Sequence[str]]) -> dict[str, Sequence[str]]:
+    """
+    Put the nested data path in a metadata dict.
+    Like get_nested, use a list if a path segment has a dot in it.
```

Comment:
since the type is sequence, should we change the "list" word in the docstring?

Reply:
Changed in a dev docs branch, PR coming soon!